File: app/src/main/java/com/andraganoid/verymuchtodo/auth/login/LoginFragment.java
package com.andraganoid.verymuchtodo.auth.login;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;

import com.andraganoid.verymuchtodo.R;
import com.andraganoid.verymuchtodo.auth.AuthBaseFragment;
import com.andraganoid.verymuchtodo.auth.register.RegisterFragment;
import com.andraganoid.verymuchtodo.databinding.LoginFragmentBinding;


public class LoginFragment extends AuthBaseFragment implements LoginClicker {

    private LoginFragmentBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(
                inflater,
                R.layout.login_fragment,
                container,
                false);
        binding.setClicker(this);
        binding.setViewModel(mainViewModel);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        main.leave = true;
        showKeyboard(binding.signInEmail);
    }

    @Override
    public void onLoginConfirm() {
        mainViewModel.userMail = "draganstojanov@mymail.com";//xxx
        mainViewModel.userPass = "testtest";
        if (!TextUtils.isEmpty(mainViewModel.userMail)
                && !TextUtils.isEmpty(mainViewModel.userPass)) {
            mainViewModel.mAuth.signInWithEmailAndPassword(
                    mainViewModel.userMail,
                    mainViewModel.userPass)
                    .addOnCompleteListener(main, task -> {
                        if (task.isSuccessful()) {
                            main.setUser(mainViewModel.mAuth.getCurrentUser());
                        } else {
                            showSnackbar(getString(R.string.auth_failed) + "\n" + task.getException().getMessage());
                        }
                    });
        }
    }

    @Override
    public void onNotRegistered() {
        main.navigateToFragment(new RegisterFragment());
    }
}

File: app/src/main/java/com/andraganoid/verymuchtodo/auth/login/LoginFragment.kt
package com.andraganoid.verymuchtodo.auth.login

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.Observer
import androidx.navigation.fragment.findNavController
import com.andraganoid.verymuchtodo.R
import com.andraganoid.verymuchtodo.auth.BaseAuthFragment
import com.andraganoid.verymuchtodo.databinding.FragmentLoginBinding
import com.andraganoid.verymuchtodo.ktodo.TodoActivity
import com.andraganoid.verymuchtodo.util.isValidEmail
import com.andraganoid.verymuchtodo.util.isValidPassword
import kotlinx.android.synthetic.main.fragment_login.*
import org.koin.android.viewmodel.ext.android.viewModel


class LoginFragment : BaseAuthFragment() {

    // private val viewModel: AuthViewModel by sharedViewModel()
    private val viewModel: LoginViewModel by viewModel()
    private lateinit var binding: FragmentLoginBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        // viewModel.showMessage(null)
        viewModel.loaderState.observe(viewLifecycleOwner, Observer { loaderState(it) })
        viewModel.loginState.observe(viewLifecycleOwner, Observer { loggedIn ->
            if (loggedIn) {
                Log.d("WEWEWE", loggedIn.toString())

                val todoIntent = Intent(main, TodoActivity::class.java)
                startActivity(todoIntent)
            }
        })
        viewModel.message.observe(viewLifecycleOwner, Observer { message ->
            if (message != null) {
                showMessage(message)
            }
        })

        binding = FragmentLoginBinding.inflate(inflater, container, false)//tod mozda izbaciti viewbinding
        binding.viewModel = viewModel
        binding.fragment = this
        return binding.root
    }


    fun submitLogin() {
        val mail = userMailEt.text.toString()
        val pass = userPassEt.text.toString()
        val messageArray = arrayListOf<String>()

        if (!mail.isValidEmail()) {
            messageArray.add(getString(R.string.mail_not_valid))
        }
        if (!pass.isValidPassword()) {
            messageArray.add(getString(R.string.password_not_valid))
        }
        hideKeyboard()
        if (messageArray.size > 0) {
            viewModel.showMessage(messageArray)
        } else {
            viewModel.login(mail, pass)
        }
    }

    fun notRegistered() {
        val action = LoginFragmentDirections.actionLoginFragmentToRegisterFragment()
        findNavController().navigate(action)
    }

    fun forgotPassword() {
        hideKeyboard()
        val mail = userMailEt.text.toString()
        if (mail.isValidEmail()) {
            viewModel.resetPassword(mail)
        } else {
            viewModel.showMessage(R.string.mail_not_valid)
        }
    }


}
